grammar.ebnf

```diff
@@ -5,16 +5,23 @@ entry                ::= comment
                       ;
 
 comment              ::= '#' .*;
-section              ::= '[[' keyword ']]';
+section              ::= '[[' __ keyword __ ']]';
+
 __                   ::= [ \t]*;
 NL                   ::= [\r\n]+;
+identifier-head      ::= [a-zA-Z_.?-];
+identifier-tail      ::= [a-zA-Z0-9_.?-];
+keyword-head         ::= [^=|#/{}\[\]()0-9];
+keyword-tail         ::= [^=|#/{}\[\]() ];
+keyword-last         ::= [^=|#/{}\[\]()];
 
-identifier           ::= [a-zA-Z_.?-] ([a-zA-Z0-9_.?-])*;
+identifier           ::= identifier-head (identifier-tail)*;
```

Comment:
it's much more readable that way and consistent with `keyword-` chars.  it makes it clear to the reader that there are different set of characters involved.